src/solace_ai_connector/components/general/db/mongo/mongo_insert.py

```
@@ -1,21 +1,97 @@
+            "name": "data_types",
+            "required": False,
+            "description": "Key value pairs to specify the data types for each field in the data. Used for non-JSON types like Date. Supports nested dotted names",
+        },
+])
 
 
 class MongoDBInsertComponent(MongoDBBaseComponent):
     """Component for handling MongoDB database operations."""
 
     def __init__(self, **kwargs):
         super().__init__(info, **kwargs)
+        self.data_types_map = self.get_config("data_types")
+        if self.data_types_map:
+            if not isinstance(self.data_types_map, dict):
+                raise ValueError(
+                    "Invalid data types provided for MongoDB insert. Expected a dictionary."
+                )
+            for key, field_type in self.data_types_map.items():
+                if not isinstance(key, str) or not isinstance(field_type, str):
+                    raise ValueError(
+                        "Invalid data types provided for MongoDB insert. Expected a dictionary with string key and value."
+                    )
+            
 
     def invoke(self, message, data):
-        if not data:
+        if not data or not isinstance(data, dict) and not isinstance(data, list):
             raise ValueError(
                 "Invalid data provided for MongoDB insert. Expected a dictionary or a list of dictionary."
             )
+        
+        if self.data_types_map:
+            for key, field_type in self.data_types_map.items():
+                if isinstance(data, list):
+                    new_data = []
+                    for item in data:
+                        new_data.append(self._convert_data_type(item, key, field_type))
+                    data = new_data
+                else:
+                    data = self._convert_data_type(data, key, field_type)
         return self.db_handler.insert_documents(data)
+    
+    def _convert_data_type(self, data, key, field_type):
+        if not key or not field_type:
```

[SUG] This condition should not happen. It means the config includes a filed with null datatype. You may need to raise an error and either breaks or continue.

---

Needed due to recursive nature of the function